src/cryptonote_core/cryptonote_tx_utils.cpp

```diff
@@ -916,14 +916,17 @@ namespace cryptonote
     return r;
   }
   //---------------------------------------------------------------
-  bool construct_tx(const account_keys& sender_account_keys, std::vector<tx_source_entry>& sources, const std::vector<tx_destination_entry>& destinations, const boost::optional<cryptonote::tx_destination_entry>& change_addr, std::vector<uint8_t> extra, transaction& tx, uint64_t unlock_time, bool is_staking, bool per_output_unlock)
+  bool construct_tx(const account_keys& sender_account_keys, std::vector<tx_source_entry>& sources, const std::vector<tx_destination_entry>& destinations, const boost::optional<cryptonote::tx_destination_entry>& change_addr, std::vector<uint8_t> extra, transaction& tx, uint64_t unlock_time, uint8_t hf_version, bool is_staking)
```

Comment:
construct_tx's semantics feels more like it shouldn't be aware of the hf_version and that it's just a building function that takes whatever parts you want to customise about the TX and construct it.

Thoughts?

Reply:
`construct_tx_and_get_tx_key` does that, `construct_tx` is more of a helper that is only called from tests (maybe it doesn't belong here?). I don't feel strongly one way or another.